### ISSUE TITLE
remove references to deleted IFinishableConsumer from documentation

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6632]
|<img alt="daf's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| daf reported|
|-|-|
|Trac ID|trac#6632|
|Type|enhancement|
|Created|2013-07-18 15:07:34Z|
|Branch|https://github.com/twisted/twisted/tree/no-ifinishableconsumer-6632|

IFinishableConsumer has been deleted as per [#5944](https://github.com/twisted/twisted/issues/5944), but the documentation still makes references to it.

https://twistedmatrix.com/documents/current/core/howto/producers.html

Attachments:

* [6632.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/163/163fb878747c8fad784e93c22ff7393a157c277f/4f1d0a59470cebdc4d2f04f665fcfc050dcd8b55.patch) (2771 bytes) - added by ewong on 2013-11-09 02:48:16Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__6632 6632
type__enhancement enhancement
reporter__daf daf
priority__normal normal
milestone__ 
branch__branches_no_ifinishableconsumer_6632 branches/no-ifinishableconsumer-6632
branch_author__itamarst itamarst
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1374160054000000 1374160054000000
changetime__1384698510000000 1384698510000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|<img alt="ewong's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|ewong set owner to ewong<br>ewong set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="ewong's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|ewong removed owner<br>ewong set status to `new`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_4"></a>|@itamarst commented|
|-|-|

(In [40771]) Branching to 'no-ifinishableconsumer-6632'

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_5"></a>|@itamarst commented|
|-|-|

Thanks for the patch!

It's missing a news file (see https://twistedmatrix.com/trac/wiki/ReviewProcess#Newsfiles), but in this case it'd just be a .misc and I'll do it. I will also run buildbot to see if it catches any errors, and if not will merge.

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_6"></a>|@itamarst commented|
|-|-|

OK, looks good.

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_7"></a>|@itamarst set status to `closed`|
|-|-|

(In [40774]) Merge no-ifinishableconsumer-6632.

Author: ewong
Reviewers: itamar
Fixes: [#6632](https://github.com/twisted/twisted/issues/6632)

Remove references to no longer existing interface in the producers howto.
